### PR TITLE
Disclaimer pop up

### PR DESCRIPTION
**Pull request by leenlab2**
Changed the disclaimer into a dismissable popup box right below the menu bar on all pages.

## Motivation and Context

The disclaimer used to take up too much unnecessary space on the page. This PR is part of a set of layout changes detailed in Issue #1218.

## Your Changes

**Description**:

- Removed the old disclaimer
- Resized container to account for removed disclaimer
- Created a new Disclaimer react component
- Inserted that component into `<Container />`, `<Grid />`, and `Generate`.
- Increased the `z-index` of `.overlay` to hide the disclaimer
- Changed link colors on the disclaimer to account for the different background color
- Added a 'Do Not Show Again' checkbox to stop the disclaimer from showing up every time

**Type of change** (select all that apply):

- [x] Other: User interface changes

## Testing

Manual testing in browser.

## Questions and Comments (if applicable)

None!

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the CircleCI tests have
  passed.

### CODE REVIEW

**Comment by leenlab2** [comment]
Comparison screenshots to showcase changes:

Before
image

After
image

**Comment by leenlab2** [comment]
Alright, I'll get working on it! I used ids instead of classes for some since they were already present in the css file (eg #disclaimerDiv), but I'll work on improving that.

File: js/components/common/Disclaimer.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+/**
+ * A React component representing the disclaimer popup
+ */
+export default class Disclaimer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dontShowDisclaimer: localStorage.getItem('DontShowDisclaimer')
```
Comment (david-yz-liu) [inline js/components/common/Disclaimer.js:10]: The names here could be better. The state name should be something standard like `hidden`, and the local storage name a corresponding `hide-disclaimer`.

File: js/components/common/Disclaimer.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        }
+    }
+
+    handleClose = () => {
```
Comment (david-yz-liu) [inline js/components/common/Disclaimer.js:14]: Avoid modifying the DOM directly, since we want to use React to do this for us. This method should set the "hide disclaimer" state to `true`. You don't need the `dismiss` class or styling, I don't think the transition is worth it.

File: js/components/common/Disclaimer.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        let disclaimer = document.getElementById("disclaimerDiv");
+        disclaimer.classList.toggle("dismiss");
+    }
+
+    handleCheck = () => {
```
Comment (david-yz-liu) [inline js/components/common/Disclaimer.js:19]: The checkbox can be both selected and deselected. So you need to access the event object to see whether the box is checked or unchecked, as this determines what to set the local storage value to.

File: js/components/common/Disclaimer.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        localStorage.setItem('DontShowDisclaimer', 'true');
+    }
+
+    render() {
+        const timetable = <a href="https://timetable.iit.artsci.utoronto.ca/">Official Timetable</a>;
+        const calendar = <a href="http://calendar.artsci.utoronto.ca/">Calendar</a>;
```
Comment (david-yz-liu) [inline js/components/common/Disclaimer.js:25]: It looks like the new permanent URL is `https://artsci.calendar.utoronto.ca/`, so please update to use this. Also, please change "Calendar" to "Academic Calendar".

File: js/components/common/Disclaimer.js (hunk 1 above)
Comment (david-yz-liu) [reply js/components/common/Disclaimer.js:10]: Oh also, do the conversion to a boolean here rather than in the `render`. Logically, this state should be a bool, not a string.

File: js/components/common/Disclaimer.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                    <label>
+                        <input type="checkbox" id="disclaimerCheck" class="dont-show-checkbox" onClick={this.handleCheck} />
+                        Do not show this again
+                    </label>
+                    <button id="closeDisclaimer" class="close-banner" onClick={this.handleClose}>&times;</button>
+                </div>
+            )
+        }
+    }
+}
```
Comment (david-yz-liu) [inline js/components/common/Disclaimer.js:44]: Make sure to end this file with a blank line.

File: js/components/common/Disclaimer.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                    <p><b>DISCLAIMER:</b> Both the {timetable} and {calendar} take precedence over the information
+                    presented here. It's important that you double-check your course selection, prerequisites,
+                    and your program plans. Some graph edges may represent a corequisite rather than a prerequisite.</p>
+                    <label>
+                        <input type="checkbox" id="disclaimerCheck" class="dont-show-checkbox" onClick={this.handleCheck} />
+                        Do not show this again
+                    </label>
+                    <button id="closeDisclaimer" class="close-banner" onClick={this.handleClose}>&times;</button>
+                </div>
+            )
```
Comment (david-yz-liu) [inline js/components/common/Disclaimer.js:41]: Add a semicolon here.

File: js/components/common/Disclaimer.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+    render() {
+        const timetable = <a href="https://timetable.iit.artsci.utoronto.ca/">Official Timetable</a>;
+        const calendar = <a href="http://calendar.artsci.utoronto.ca/">Calendar</a>;
+
+        if (this.state.dontShowDisclaimer === 'true') {
+            return (null);
+        } else {
+            return (
+                <div id="disclaimerDiv" className="popup-banner">
```
Comment (david-yz-liu) [inline js/components/common/Disclaimer.js:31]: Since you've added the classes, I suspect that the ids here and for the inner input/button elements are no longer necessary, and you can remove them.

File: js/components/grid/grid.js.jsx [hunk 8]
```diff
@@ -164,8 +165,10 @@ class Grid extends React.Component {
 
   render() {
     const updatedList = this.state.hoveredLecture ? this.state.selectedLectures.concat(this.state.hoveredLecture) : this.state.selectedLectures;
+
```
Comment (david-yz-liu) [inline js/components/grid/grid.js.jsx:168]: Revert this change

File: style/app.scss [hunk 9]
```diff
@@ -233,10 +233,51 @@ a:active
...
 }
 
-#disclaimerDiv
+.popup-banner
 {
-  padding   : 0px 1em 0px 1em;
-  font-size : 11pt;
+  font-size        : 11pt;
+  position         : absolute;
+  background-color : #ffdd61;
```
Comment (david-yz-liu) [inline style/app.scss:240]: I'd prefer a yellow that's slightly less saturated (i.e., closer to white).

File: style/app.scss [hunk 10]
```diff
@@ -233,10 +233,51 @@ a:active
...
+.popup-banner.dismiss
+{
+  transition : visibility 0s 0.5s, opacity 0.5s linear;
+  visibility : hidden;
+  opacity    : 0;
+}
+
+.popup-banner a
+{
+  color : #af16af;
```
Comment (david-yz-liu) [inline style/app.scss:254]: Don't set a different colour here. Instead, use `text-decoration: underline` (in fact, we should have `text-decoration: underline` active by default for all `a` tags, but that's a different story).

File: style/app.scss [hunk 11]
```diff
@@ -233,10 +233,51 @@ a:active
...
+}
+
+.popup-banner label
+{
+  font-weight : lighter;
+  color       : #696969;
+  position    : absolute;
```
Comment (david-yz-liu) [inline style/app.scss:261]: Don't overuse absolute positioning. In this particular case, you shouldn't assume that there will be empty space in the bottom-right corner of the banner, as this depends on screen size. Use a different layout here. For example, put the "X" in the top-right corner, and the checkbox at the bottom.

File: style/app.scss [hunk 12]
```diff
@@ -233,10 +233,51 @@ a:active
...
+  left        : 82%;
+  top         : 30px;
+}
+
+.close-banner
+{
+  cursor      : pointer;
```
Comment (david-yz-liu) [inline style/app.scss:268]: This rule is redundant

File: style/app.scss [hunk 13]
```diff
@@ -233,10 +233,51 @@ a:active
...
+}
+
+.popup-banner a
+{
+  color : #af16af;
+}
+
+.popup-banner label
+{
+  font-weight : lighter;
```
Comment (david-yz-liu) [inline style/app.scss:259]: In general, do not introduce new styles for text or other elements unless there's a good reason we need something new. In this case, you should remove this rule and the new color.

**Review by david-yz-liu — COMMENTED** [review]
In addition to the comments I left inline, I'd like you to experiment with different placements of this notification. I'm not sure a full-width banner is the best approach. You can try making the banner less wide and either centered at the top, centered at the bottom, or in a corner (I've seen bottom-left or bottom-right on a few websites). Possibly with a stronger border and/or rounded edges to make it more obvious that it's "on top of" the rest of the page.

**Comment by leenlab2** [comment]
I fixed the issues you asked for, and changed the disclaimer design to look like this:
image

**Review by david-yz-liu — COMMENTED** [review]
@leenlab2 very nice work! I just left a few more comments. Almost there!

**Comment by leenlab2** [comment]
Alright, should be done now! Let me know if there is anything else I need to fix.
